#25766 Move the Quicktime generation settings into a hook

kporangehat opened the pull request:
Allows studios to easily override the default output settings. Also adds handling for the knob name change in Nuke 9.0 from `codec` to `meta_codec`

Commit:
#25766 Move the Quicktime generation settings into a hook
Allows studios to easily override the default output settings. Also adds handling for the knob name change in Nuke 9.0 from `codec` to `meta_codec`

manneohrstrom commented on info.yml:
```diff
@@ -52,6 +52,14 @@ configuration:
+                     a Quicktime. The return value is a dictionary where the
+                     keys are knob names for the write node, and the values are
+                     the corresponding node values."
+        default_value: nuke_quickdailies_quicktime_settings
```
Please use new hook syntax: '{self}/quicktime_settings.py'

manneohrstrom commented on hooks/nuke_quickdailies_quicktime_settings.py:
```diff
@@ -0,0 +1,43 @@
+    """
+    Allows modifying default settings for Quicktime generation.
+    """
+    def execute(self, **kwargs):
```
I think you should pass the nuke node name in here - this would allow for a studio to write code where they can resolve the node object in nuke and perform some custom logic.

robblau replied:
Also, a shorter name here would be good.  GetQuicktimeSettings?

manneohrstrom commented:
Similar to my comment here - https://github.com/shotgunsoftware/tk-multi-reviewsubmission/pull/5#issuecomment-141020518 - i think the hook should be named codec settings and the method inside should be called `get_quicktime_settings()`. We should move away from using `execute()` in hooks, this is the deprecated syntax.

Commit:
CR notes addressed
- added setting to upload the movie to Shotgun (false by default for backwards compatibility)
- added new codec handling for Nuke 9 on linux
- made check for version 9 read a little more intuitive

Commit:
renamed hook to codec_settings

kporangehat commented:
closed by 705d4c717463cf8f57d971c35345b71b32039024
